Sources/SectionScrubber.swift

```diff
@@ -167,82 +196,89 @@ public class SectionScrubber: UIView {
-
         self.sectionLabelState = gesture.state == .Ended ? .Hidden : .Visible
 
         if gesture.state == .Began || gesture.state == .Changed || gesture.state == .Ended {
-            let translation = gesture.translationInView(self)
-            if gesture.state == .Began {
-                self.startOffset = collectionView.contentOffset.y
-            }
+            let locationInCollectionView = gesture.locationInView(collectionView)
+            let locationInWindow = collectionView.convertPoint(locationInCollectionView, toView: window)
+            let location = locationInWindow.y - (self.adjustedContainerOrigin + collectionView.contentInset.top)
 
-            let containerHeight = self.containingViewFrame.height - self.viewHeight
-            let totalHeight = collectionView.contentSize.height - self.containingViewFrame.height
-            let basePercentage = (self.startOffset - originalYOffset) / totalHeight
-            let currentPercentage = translation.y / containerHeight
-            var percentageInView = basePercentage + currentPercentage
+            let gesturePercentage = self.boundedPercentage(location / self.adjustedContainerBoundsHeight)
+            let y = (self.adjustedContainerHeight * gesturePercentage) - collectionView.contentInset.top
+            collectionView.setContentOffset(CGPoint(x: collectionView.contentOffset.x, y: y), animated: false)
 
-            if percentageInView < 0 {
-                percentageInView = 0
-            }
+            self.userInteractionOnScrollViewDetected()
+        }
+    }
 
-            if percentageInView > 1 {
-                percentageInView = 1
-            }
+    private func boundedPercentage(percentage: CGFloat) -> CGFloat {
+        var newPercentage = percentage
 
-            let yOffset = (totalHeight * percentageInView) + originalYOffset
-            collectionView.setContentOffset(CGPoint(x: collectionView.contentOffset.x, y: yOffset), animated: false)
-        }
+        newPercentage = max(newPercentage, 0.0)
+        newPercentage = min(newPercentage, 1.0)
+        return newPercentage
     }
 
     func handleLongPress(gestureRecognizer: UILongPressGestureRecognizer) {
         self.sectionLabelState = gestureRecognizer.state == .Ended ? .Hidden : .Visible
         self.userInteractionOnScrollViewDetected()
     }
 
-    private func setSectionLabelFrame() {
-        let rightOffset = self.sectionLabelState == .Visible ? SectionLabel.RightOffsetForActiveSectionLabel : SectionLabel.RightOffsetForInactiveSectionLabel
-        self.sectionLabel.frame = CGRectMake(self.frame.width - rightOffset - self.sectionLabel.sectionlabelWidth, 0, self.sectionLabel.sectionlabelWidth, viewHeight)
-    }
-
     private func animateScrubberState(state: VisibilityState, animated: Bool) {
         let duration = animated ? 0.2 : 0.0
+        switch state {
+        case .Visible:
+            self.scrubberImageRightConstraint.constant = -6
```

Comment:
Constant looks like a magic number, could you make a constant for it and put it on top of the file?